crates/qt-build-utils/src/parse_cflags.rs

```diff
@@ -167,13 +167,18 @@ pub(crate) fn parse_libs_cflags(name: &str, link_args: &[u8]) {
                     if let (Some(dir), Some(file_name), Ok(target)) =
                         (path.parent(), path.file_name(), &target)
                     {
-                        match extract_lib_from_filename(target, &file_name.to_string_lossy()) {
-                            Some(lib_basename) => {
-                                println!("cargo:rustc-link-search={}", dir.display());
-                                println!("cargo:rustc-link-lib={lib_basename}");
-                            }
-                            None => {
-                                println!("cargo:warning=File path {} found in .prl file for {name}, but could not extract library base name to pass to linker command line", path.display());
+                        if file_name.to_string_lossy().ends_with(".o") {
+                            #[cfg(feature = "include_qt_objects")]
+                            _builder.object(path);
```

Comment:
Oh this is an interesting hack; I didn't realize the cc crate could do this. This should fix linking to Qt6Quick statically (#513). Why is this behind a feature flag though? IIUC this should always be done.

Please add a TODO comment linking to https://github.com/rust-lang/rust/issues/99427#issuecomment-1562092085 and explaining that this should be replaced when rustc has a CLI for linking object files.

Reply:
When rustc has support for linking object files, we should contribute that upstream to the pkg-config crate which is where this code came from. Add a dependency on the `cc` crate wouldn't work for that, but it's fine for this temporary hack.

Reply:
The reason I put this behind a feature flag is for integration with other build systems like cmake and qmake.

qmake when building for wasm in particular, pulls in a lot of dependencies automatically resulting in lots of duplicate symbol errors at the linking stage when this feature is turned on.

Reply:
I have been playing with merging this branch together with #513 and got it to link by deduplicating the `.o` files (plus a quick hack around https://github.com/KDAB/cxx-qt/pull/600/files/cebe9711c434f336d15cc98512d79cccfeeb47f4#diff-e4bb99471e34daf6f9630c7df936bd268f023426f24f3f39ba7658079ff2ffe9). I made a PR for your fork with this fix: https://github.com/jimmyvanhest/cxx-qt/pull/1 Please test that and see if it avoids the duplicate symbol link errors you were encountering before.